lntest/node.go

```diff
@@ -661,18 +668,46 @@ func (hn *HarnessNode) start(lndBinary string, lndError chan<- error) error {
 	return hn.initLightningClient(conn)
 }
 
+// WaitUntilLeader attempts to finish the start procedure by initiating an RPC
+// connection and setting up the wallet unlocker client. This is needed when
+// a node that has recently been started was waiting to become the leader and
+// we're at the point when we expect that it is the leader now (awaiting unlock).
+func (hn *HarnessNode) WaitUntilLeader(timeout time.Duration) error {
+	var (
+		conn    *grpc.ClientConn
+		connErr error
+	)
+	if err := wait.NoError(func() error {
```

Comment:
Again here it could make an explicit query to check if it's leader or not, instead of attempting the RPC connection check, which I think will always succeed now w/ the latest wallet unlocker unification?

Reply:
With the new RPC state service we'll wait until the state is advanced from "waiting to start". PTAL